Week3/homework/server.js

```diff
@@ -0,0 +1,190 @@
+    if (err) console.log(err);
+    con.query(
+      `insert into users values (?,?,?)`,
+      [result.length + 1, req.body.username, req.body.password],
+      (err, data) => {
+        if (err) {
+          res.render('error', { data: 'error' });
+        } else {
+          res.render('index');
+        }
+        con.end();
+      },
+    );
+  });
+});
+
+// log in => get user todo list
+app.get('/login/user', urlParser, (req, res) => {
+  con.query(
+    `select username,password from users where username= '${req.query.username}'`,
+    (err, data) => {
+      if (err) res.render('login', { error: err });
+      if (!data[0]) {
+        res.render('login', { error: `username or password not correct` });
+      } else if (data[0].password !== req.query.password) {
+        res.redirect('login');
+      } else {
+        con.query(
+          ` select id,name from list where uid = (select id from users where username='${req.query.username}');`,
+          (err, option) => {
+            res.render('user', {
+              username: req.query.username,
+              password: req.query.password,
+              items: option,
+            });
+            con.end();
+          },
+        );
+      }
+    },
+  );
+});
+
+//  add list
+app.post('/user/add/list', urlParser, (req, res) => {
+  con.query(`select id from users where username = '${req.body.username}' ;`, (err, response) => {
+    if (err) console.log(err);
+    con.query(
+      `insert into list (name,uid,deadline,reminderid) values (?,?,?,?)`,
+      [req.body.name, response[0].id, req.body.deadline, req.body.reminder],
+      (err, data) => {
+        if (err) console.log(err);
+        con.query(
+          ` select id,name from list where uid = (select id from users where username='${req.body.username}');`,
+          (err, option) => {
+            res.render('user', {
+              username: req.body.username,
+              password: req.body.password,
+              items: option,
+            });
+            con.end();
+          },
+        );
+      },
+    );
+  });
+});
+
+// add item
+app.post('/user/add/item', urlParser, (req, res) => {
+  con.query(
+    `insert into items (description,listid) values (?,?)`,
+    [req.body.description, req.body.list],
+    (err, result) => {
+      if (err) console.log(err);
+      con.query(
+        ` select id,name from list where uid = (select id from users where username='${req.body.username}');`,
+        (err, option) => {
+          res.render('user', {
+            username: req.body.username,
+            password: req.body.password,
+            items: option,
+          });
+          con.end();
+        },
+      );
+    },
+  );
+});
+
+//  show lists
+app.get('/user/lists', urlParser, (req, res) => {
+  con.query(
+    `select * from items inner join list on items.listid = list.id where uid in (select id from users where username='${req.query.username}') group by name;`,
```

Comment:
this query is not working, you can't group by name , and you don't need too, username should be uniq in your case otherwise that means you bring list for an other user.
By the way, do you want to bring list or items ? because if I don't have item then I don't bring anything..

Avoid using ${req.query.username} directly in the query but instead use prepared statement ? otherwise your code becomes vulnerable to attack.